### ISSUE TITLE
How to parse a cel expression? 

### ISSUE THREAD

**Issue by HoneyryderChuck** [google/cel-cpp#125]
Greetings,

I was browsing through the code, but could find the module responsible for parsing and checking cel expression. I could only find what's under eval/compile, but examples only parse syntax très of expressions in text format.

Is the functionality available? I'm interested in building a ruby library r wrapping calls to this lib and offering the same api as cel-go

**Comment by kyessenov** [comment]
Yes, it's available. https://github.com/google/cel-cpp/blob/master/conformance/server.cc#L54 is a good starting point.

**Comment by HoneyryderChuck** [comment]
I investigated this today a bit further, and I think it's going to be hard... I can't seem to build via bazel successfully, and there's little precedent of integrating a c++ library as a ruby extension without a more "conventional" build toolchain such as `make` and include headers to link to C. I'm basing myself on how google `grpc` itself builds as a ruby gem, and the steps it follows to link libgrpc C++ code.

Does google have plans to release multi-lingo support for CEL, like it does for `protobuf` and `grpc`? It'd be great to know whether there's a plan, so that maybe waiting out and hacking around `cel` (or even wait for it) can be considered. Currently there's only c++ and go as options, and neither is an adopted tech stack where I need it.

**Comment by kyessenov** [comment]
What kind of issue did you encounter with `bazel`? It's difficult to maintain non-bazel compiler toolchain for C++ for a lot of Google code. There are some piece-meal CMake integrations, but they are always supplementary, not primary.
 
I think it's possible that a Java version would be also open sourced. Is that something you'd use?

**Comment by HoneyryderChuck** [comment]
> What kind of issue did you encounter with bazel?

Somewhere during call invokes, I get to an error while downloading a package:

```
ERROR: /Users/dev/dev/cel-cpp/eval/public/BUILD:172:1: //eval/public:builtin_func_registrar depends on @com_googlesource_code_re2//:re2 in repository @com_googlesource_code_re2 which failed to fetch. no such package '@com_googlesource_code_re2//': java.io.IOException: Error downloading [https://github.com/google/re2/archive/master.zip] to /private/var/tmp/_bazel_dev/7f1dcaa7261468e62aa85becde6b4890/external/com_googlesource_code_re2/master.zip: GET returned 404 Not Found
```

I'd still have issues setting it up for a native extension though, so that'd be just the 1st hurdle to go across. I understand it's not really my call to make a make-based project though, would just help this particular scenario perhaps.

> I think it's possible that a Java version would be also open sourced. Is that something you'd use?

I'd consider it. I could still keep using ruby via `jruby`. I'd might bump into issues using it protobufs with it (`google/protobuf` doesn't work for `jruby`), but I could work around it using java protobuf.

It'd be great for `cel` to release multiple wrappers though, like it's done for `grpc` or `protobuf` 🙂  Obviously not my call to make, so thanks for your valuable input 🙏  .

**Comment by kyessenov** [comment]
Thanks for reporting the build problem! google/re2 renamed its main branch which caused the build error. 

We try to keep the set of dependencies small here (https://github.com/google/cel-cpp/blob/master/bazel/deps.bzl), so it's realistic to create a CMake build, provided all dependencies support CMake. But it's difficult to keep it in sync.

**Comment by HoneyryderChuck** [comment]
Just sharing that I've released a [pure ruby cel library](https://gitlab.com/honeyryderchuck/cel-ruby).
